odl/util/numerics.py

```diff
@@ -297,6 +303,593 @@ def fast_1d_tensor_mult(ndarr, onedim_arrs, axes=None, out=None):
+            n_pad_r = len(range(istop, n_new))
+
+            # Handle some error scenarios with illegal lengths
+            if pad_mode == 'order0' and n_orig == 0:
+                raise ValueError('in axis {}: need at least 1 value for '
+                                 'order 0 padding, got 0'
+                                 ''.format(i))
+
+            if pad_mode == 'order1' and n_orig == 1:
+                raise ValueError('in axis {}: need at least 2 values for '
+                                 'order 1 padding, got 1'
+                                 ''.format(i))
+
+            for lr, pad_len in [('left', n_pad_l), ('right', n_pad_r)]:
+                if pad_mode == 'periodic' and pad_len > n_orig:
+                    raise ValueError('in axis {}: {} padding length {} '
+                                     'exceeds original array size {}; this is '
+                                     'not allowed for periodic padding'
+                                     ''.format(i, lr, pad_len, n_orig))
+
+                elif pad_mode == 'symmetric' and pad_len >= n_orig:
+                    raise ValueError('in axis {}: {} padding length {} '
+                                     'larger or equal to the original array '
+                                     'size {}; this is not allowed for '
+                                     'symmetric padding'
+                                     ''.format(i, lr, pad_len, n_orig))
+
+            arr_slc.append(slice(None))
+            out_slc.append(slice(istart, istop))
+            pad_l_slc.append(slice(istart))
+            pad_r_slc.append(slice(istop, None))
+
+        else:
+            arr_slc.append(slice(None))
+            out_slc.append(slice(None))
+            pad_l_slc.append(slice(0))
+            pad_r_slc.append(slice(0))
+
+
+def resize_array(arr, newshp, frac_left=None, num_left=None,
+                 pad_mode='constant', pad_const=0, direction='forward',
+                 out=None):
+    """Return the resized version of ``arr`` with shape ``newshp``.
+
+    In axes where ``newshp > arr.shape``, padding is applied according
+    to the supplied options.
+    Where ``newshp < arr.shape``, the array is cropped to the new
+    size.
+
+    Parameters
+    ----------
+    arr : array-like
+        Array to be resized.
+    newshp : sequence of int
+        Desired shape of the output array.
+    frac_left : float or sequence of float, optional
+        Number between 0 and 1 (inclusive) that determines which fraction
+        of the addition/removal is performed on the "left" side. If
+        ``frac_left`` times the shape difference is not integer, the
+        number of entries added to the left is gained by rounding to the
+        nearest integer. The default ``None`` is equivalent to 0.5.
+        This option can be specified per axis with a sequence.
+        Cannot be combined with ``num_left``.
+    num_left : sequence of int, optional
```

Comment:
`num_right` can be caluclated from the shapes and `num_left`, but sure, I can add it for convenience. `left` is something I chose since it's intuitively clear in 1D, but it's less so in higher dimensions. Alternatives?
- left/right
- before/after 
- top/bottom
- ?